### PR TITLE
modbus-go: fix logical exceptions

### PR DESCRIPTION
**Pull request by soyoo**
Both certification and private key must be provided

soyoo@aliyun.com
/assign @sids-b

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: mappers/modbus-go/config.go [hunk 1]
```diff
@@ -77,8 +77,7 @@ func (c *Config) parseFlags() error {
 	pflag.StringVar(&c.Mqtt.PrivateKey, "mqtt-priviatekey", c.Mqtt.PrivateKey, "private key file path")
 	pflag.Parse()
 
-	if (c.Mqtt.Cert != "" && c.Mqtt.PrivateKey == "") ||
-		(c.Mqtt.Cert == "" && c.Mqtt.PrivateKey != "") {
+	if c.Mqtt.Cert == "" || c.Mqtt.PrivateKey == "" {
```
Comment (fisherxu) [inline mappers/modbus-go/config.go:80]: If c.Mqtt.Cert and c.Mqtt.PrivateKey are both empty, maybe we shouldn't return error? @sailorvii

Comment (soyoo) [reply mappers/modbus-go/config.go:80]: If it is like you say, it is better to change error message.

Comment (fisherxu) [reply mappers/modbus-go/config.go:80]: We should support unsecure connection?

Comment (sailorvii) [reply mappers/modbus-go/config.go:80]: For the Mqtt broker could be external, the authentication method depends on what the broker requires.
So I think it's better to support different methods.

**Comment by daixiang0** [comment]
@soyoo now we move modbus-go into kubeedge/mappers-go repo, could you post this here?

**Comment by soyoo** [comment]
> @soyoo now we move modbus-go into kubeedge/mappers-go repo, could you post this here?

How? copy the change to kubeedge/mappers-go repo?

**Comment by fisherxu** [comment]
> > @soyoo now we move modbus-go into kubeedge/mappers-go repo, could you post this here?
> 
> How? copy the change to kubeedge/mappers-go repo?

Yes, you can open new pr in  kubeedge/mappers-go repo

**Comment by daixiang0** [comment]
/close 
in favor of https://github.com/kubeedge/mappers-go/pull/5

**Pull request closed without merging**